Automattic/sensei
# Rename restrict content block and its menu

Part of #3934

### Changes proposed in this Pull Request

* Renames 'Restricted Content' block to 'Conditional Content' block.
* Renames the dropdown menu from 'Visibility' to 'Visible when'
* Also fixed an unrelated issue where the block had big margins only in editor in 2021 theme.
* In Seedlet theme in the frontend, the block appears to have left and right padding. But the same also happens with the Group block so I think that we should ignore it as a theme issue.

### Testing instructions

* Do a smoke test of the block and observe that it works as it should.
* Check that the spaces of the block appear correctly on 2021.
